orchestra/utils/tests/test_task_lifecycle.py

```diff
@@ -769,3 +771,35 @@ def test_next_todo_with_earlier_start_time(self, mock_timezone):
                 # should select the todo with a deadline
                 self.assertEqual(next_todo_start, DEADLINE1_DATETIME)
                 self.assertEqual(t['should_be_active'], False)
+
+
+class EndProjectTestCase(OrchestraTransactionTestCase):
+    def setUp(self):
+        super().setUp()
+        setup_models(self)
+        path = 'orchestra.tests.helpers.workflow.abortion_cleanup_function'
+        self.abort_completion_function = {
+            'path': path,
+            'kwargs': {
+                'some_key': 'some_value'
+            }
+        }
+
+    @patch('orchestra.utils.task_lifecycle.archive_project_slack_group')
+    @patch('orchestra.utils.task_lifecycle._call_abort_completion_function')
+    def test_end_project_calls_abort_completion_fn(
+            self, mock_call_abort_fn, mock_archive_project_slack_group):
+        project = self.projects['test_human_and_machine']
+        project.workflow_version.abort_completion_function = (
+            self.abort_completion_function)
+        end_project(project.id)
+        mock_call_abort_fn.assert_called_with(project)
```

Comment:
Since `_call_abort_completion_function` is a `_private_function` to this module, I wouldn't add a unit test to ensure the function is implemented a certain way. Instead, I'd call `end_project`, not mock `_call_abort_completion_function` in this test, and add the mocks/asserts you have in `in test_call_abort_completion_function` to the current test. Then I'd delete `test_call_abort_completion_function`.

In total, there's be a single test in this class. Let me know if that sounds sane.